Cannot find module 'json.macro/types' or its corresponding type declarations

### Summary

I'm using remirror in a typescript project. When I build this project with `tsc`, I get the following error:

```
$ tsc
node_modules/remirror/dist/declarations/src/index.d.ts:1:38 - error TS2307: Cannot find module 'json.macro/types' or its corresponding type declarations.

1 import type { SemanticVersion } from 'json.macro/types';
                                       ~~~~~~~~~~~~~~~~~~
```

We use the npm package `json.macro` to read the `package.json` at the build time, and that's why we only add this package in `devDependencies`.

<!-- A brief summary of the problem. -->

### Steps to reproduce

N/A

<!-- All bug reports should include detailed steps to reproduce -->

### Expected results

Not error.
<!-- Outline what you expected to see -->

### Actual results

Got error bacause `tsc` can't find `json.macro`  in the `node_modules`.
<!-- Outline the actual results -->

### Possible Solution

Copy the type from `json.macro` into the remirror repo.

<!--- Optionally suggest a fix or reason for the bug -->

--- a/packages/remirror/src/index.ts
+++ b/packages/remirror/src/index.ts
@@ -1,5 +1,63 @@
 import { getVersion } from 'json.macro';
-import type { SemanticVersion } from 'json.macro/types';
+
+/**
+ * This interface is copied from `json.macro/types` to avoid some runtime error
+ * when using remirror as a dependency.
+ *
+ * See also: https://github.com/remirror/remirror/issues/943
+ **/
+interface SemanticVersion {
+  /**
+   * The raw un-parsed version taken directly from your `package.json` file. This
+   * includes the build numbers.
+   */
+  raw: string;
+
+  /**
+   * Whether this version is loose or not.
+   */
+  loose: boolean;
+
+  /**
+   * The major version.
+   *
+   * `3.4.1-alpha.100+a.123` => `3`
+   */
+  major: number;
+
+  /**
+   * The minor version.
+   *
+   * `3.4.1-alpha.100+a.123` => `4`
+   */
+  minor: number;
+
+  /**
+   * The patch version.
+   *
+   * `3.4.1-alpha.100+a.123` => `1`
+   */
+  patch: number;
+
+  /**
+   * The version as a string without the build number.
+   */
+  version: string;
+
+  /**
+   * The build number for this version
+   *
+   * `3.4.1-alpha.100+a.123` => `['a', 123]`
+   */
+  build: readonly string[];
+
+  /**
+   * The prerelease string.
+   *
+   * `3.4.1-alpha.100+a.123` => `['alpha', 100]`
+   */
+  prerelease: ReadonlyArray<string | number>;
+}
 
 export const VERSION: SemanticVersion = getVersion(true);
 

